onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -706,3 +761,34 @@ ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProvider_Tensorrt, _In_ OrtS
   options->provider_factories.push_back(factory);
   return nullptr;
 }
+
+ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider_OpenVINO, _In_ OrtSessionOptions* options, _In_ const OrtOpenVINOProviderOptions* provider_options) {
+  auto factory = onnxruntime::CreateExecutionProviderFactory_OpenVINO(provider_options);
+  if (!factory) {
+    return OrtApis::CreateStatus(ORT_FAIL, "SessionOptionsAppendExecutionProvider_OpenVINO: Failed to load shared library");
+  }
+
+  options->provider_factories.push_back(factory);
+  return nullptr;
+}
+
+ORT_API_STATUS_IMPL(OrtSessionOptionsAppendExecutionProvider_OpenVINO, _In_ OrtSessionOptions* options, _In_ const char* device_type) {
+  OrtOpenVINOProviderOptions provider_options{device_type, false, "", 8};
```

Comment:
curious what does 8 represent here?

Reply:
It's number of threads. I think we can set it to 0 here. Openvino provider factory will default it to 8 if invalid value encountered.

---
In reply to: [521912749](https://github.com/microsoft/onnxruntime/pull/5778#discussion_r521912749) [](ancestors = 521912749)

Reply:
Yep, I looked internally and anything <=0 turns into 8. I made it 0.